### ISSUE TITLE
CS1584 and CS1658 on cref with generic type of generic type

### ISSUE THREAD

**Issue by niklaskarl** [DotNetAnalyzers/StyleCopAnalyzers#2304]
StyleCopAnalyzers incorrectly reports a cref attribute to be wrong if it references a generic type whose generic attribute is again set to a generic type.

As an example, the following code reports CS1584 and CS1658 on the cref attribute.

```
/// <summary>
/// Gets the next page from the result set.
/// </summary>
/// <returns>A <see cref="Task{Page{Activity}}"/> representing the asynchronous operation.</returns>
public async Task<Page<Activity>> GetNextPageAsync()
{
}
```

> CS1584 XML comment has syntactically incorrect cref attribute 'Task{Page{Activity}}'
> CS1658 Type parameter declaration must be an identifier not a type. See also error CS0081.

**Comment by sharwell** [comment]
Hi @niklaskarl,

Note that warnings and errors that start with `CS` are reported by the compiler and not by this project. With that said, the code sample provided does have an incorrect reference. Since it's not possible to reference an instantiation of a generic type from a documentation comment, you need to use the following instead:

```
<see cref="Task{T}"/>
```
